source/encoding.c

```diff
@@ -66,7 +66,9 @@ static const uint8_t BASE64_DECODING_TABLE[256] = {
 int aws_hex_compute_encoded_len(size_t to_encode_len, size_t *encoded_length) {
     AWS_ASSERT(encoded_length);
 
-    size_t temp = (to_encode_len << 1) + 1;
+    /* For every byte of input, there will be 2 hex chars of encoded output */
+
+    size_t temp = to_encode_len << 1;
 
     if (AWS_UNLIKELY(temp < to_encode_len)) {
```

Comment:
trivial: unrelated, but will it be easier to use `aws_mul_size_checked` directly?

Reply:
I'm trying to leave as much as possible unchanged.

I actually rewrote a bunch of them yesterday, thinking that they were always overestimating, then discovered I didn't totally understand the math, so put it back like it was, just with more comments 🤷‍♀️